sympy/matrices/matrices.py

```diff
@@ -384,15 +384,56 @@ def cofactor_matrix(self, method="berkowitz"):
                          lambda i, j: self.cofactor(i, j, method))
 
     def det(self, method="bareiss", iszerofunc=None):
-        """Computes the determinant of a matrix.  If the matrix
-        is at most 3x3, a hard-coded formula is used.
-        Otherwise, the determinant using the method `method`.
+        """Computes the determinant of a matrix.
 
+        Parameters
+        ==========
+
+        method : string, optional
+            Specifies the algorithm used for computing the matrix determinant.
+
+            If the matrix is at most 3x3, a hard-coded formula is used and the
+            specified method is ignored. Otherwise, it defaults to 
+            ``'bareiss'``.
+
+            If it is set to ``'bareiss'``, Bareiss' fraction-free algorithm will
+            be used.
+
+            If it is set to ``'berkowitz'``, Berkowitz' algorithm will be used.
+
+            Otherwise, if it is set to ``'lu'``, LU decomposition will be used.
+
+            .. note::
+                For backward compatibility, legacy keys like "bareis" ,
+                "det_lu" can still be used to indicate corresponding
+                methods.
+                And the keys are also case-insensitive for now.
+                However, we would discourage you to use the keys in the manner.
```

Comment:
Possibly reword this in an optimistic way `However, it is suggested to use the proper keys for specifying the method.` and write it in continuity inside the note (without starting a new paragraph).